### PR TITLE
GH-106360: Support very basic superblock introspection

### PR DESCRIPTION
**Pull request by markshannon**
Adds indexing and `len()` support to uop superblocks, such that this:
```Py
def testfunc(x):
    i = 0
    while i < x:
        i += 1

testfunc(100)

for offset in range(0, 100, 2):
    try:
        ex = _testinternalcapi.get_executor(testfunc.__code__, offset)
        break
    except ValueError:
        pass

for uop in ex:
    print(uop)
```
produces this
```
('SAVE_IP', 7)
('LOAD_FAST', 1)
('SAVE_IP', 8)
('LOAD_CONST', 2)
('SAVE_IP', 9)
('_GUARD_BOTH_INT', 13)
('_BINARY_OP_ADD_INT', 13)
('SAVE_IP', 11)
('STORE_FAST', 1)
('SAVE_IP', 12)
('LOAD_FAST', 1)
('LOAD_FAST', 0)
('SAVE_IP', 13)
('COMPARE_OP_INT', 18)
('SAVE_IP', 15)
```

* Issue: gh-106360

### CODE REVIEW

**Comment by markshannon** [comment]
This PR also adds `_PyOpcode_OpName` and `_PyOpcode_uop_name` to non-debug builds because they are only a few Kbs of static data, and it is nice to be able to introspect things.

File: Lib/test/test_capi/test_misc.py [hunk 1]
```diff
@@ -2415,5 +2415,28 @@ def long_loop():
...
+        ex = None
+        for offset in range(0, 100, 2):
+            try:
+                ex = _testinternalcapi.get_executor(testfunc.__code__, offset)
+                break
+            except ValueError:
+                pass
+        if ex is None:
+            return
+        self.assertIn("SET_IP", str(ex))
```
Comment (gvanrossum) [inline Lib/test/test_capi/test_misc.py]: I think once you merge main you'll see that this is now called `SAVE_IP`.

**Review by gvanrossum — APPROVED** [review]
This just needs a merge.

In the light of rising concerns over symbols starting with `_Py` still being public, maybe we should rename these or move the header file? These tables should be allowed to change on a whim.

**Comment by markshannon** [comment]
> In the light of rising concerns over symbols starting with _Py still being public.

They aren't public though. We strip `_Py` symbols from the binary, unless they are marked with the `PyAPI` macro.

**Pull request merged**